app/controllers/tags_controller.rb

```diff
@@ -0,0 +1,84 @@
+class TagsController < ApplicationController
+	before_action :set_tag , only: [:show]
+
+  def index
+
+  end
+
+  def edit
+    @tag = Tag.find(tag_params[:id])
+
+  end
+
+	def show
+
+  end
+
+  def new
+    @tag = Tag.new
+    @file = BruseFile.find(params[:file_id])
+
+  end
+
+  def create
+    
+    @file = BruseFile.find_by(:id => tag_params[:file_id])
+    tag_params[:name].split.each do |tag|
+      @tag = Tag.find_or_create_by(:name => tag)
+      @tag.bruse_files.append( @file )
+    end
+
+ 
+     respond_to do |format|
+        if @tag.save
+          format.html { redirect_to bruse_files_path, notice: 'Tag was successfully created.' }
+          format.json { render :show, status: :ok, location:  @tag.bruse_files }
+        else
+          format.html { render :new }
+          format.json { render json: @tag.errors, status: :unprocessable_entity }
+        end
+     end
+  end
+
+
+  
+
+  def update
```

Comment:
With this in mind, shouldn't the name of the tag be unique? Or is there any point in storing the tags twice?